src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -276,8 +382,13 @@ static CHIP_ERROR Resolve(ResolveContext * sdCtx, uint32_t interfaceId, chip::In
     auto err = DNSServiceCreateConnection(&sdCtx->serviceRef);
     VerifyOrReturnError(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
 
-    auto sdRefCopy = sdCtx->serviceRef; // Mandatory copy because of kDNSServiceFlagsShareConnection
-    err            = DNSServiceResolve(&sdRefCopy, kResolveFlags, interfaceId, name, type, kLocalDot, OnResolve, sdCtx);
+    // Similar to browse, will try to resolve using both the local domain and the open thread domain.
+    auto sdRefLocal = sdCtx->serviceRef; // Mandatory copy because of kDNSServiceFlagsShareConnection
```

Comment:
This is incorrect. You should resolve the domain that you got in the browse callback. There is no promise that if you do a browse in local, all your responses will be in local. Same for default.service.arpa. There is no need to do two resolves here, and in fact it's incorrect because there might be totally unrelated data that happens to match the hostname you got for local in default.service.arpa, or vice versa.

Reply:
We discussed this with you in our meeting with Boris. So we get a browse before resolve only for the pairing use cases. for operational reads/writes, we do not get a browse before a resolve and hence we had to do the same thing as browse. I know you re-iterated that what we get in the browse result can be different than what we requested. so what i could do is if the domain in the browse result is not null i.e if a browse preceded a resolve, i will use the domain from the browse result otherwise we will resolve on local domain and srp domain. would that work?

Reply:
You should never get a NULL domain in a browse callback. Are you seeing that happen?

Reply:
This needs a discussion with @bzbarsky-apple. The code right now should work for the 2 domains we care about the local and srp domain but definitely has a bug. Need to be re-written as follow up

Reply:
filed: #32681